src/mlpack/tests/knn_test.cpp

```diff
@@ -16,6 +16,18 @@
 
 using namespace mlpack;
 
+/**
+ * A couple of handful declarations for float32 testing.
+ * These will be removed when we refactor the Bounds to accept MatType.
+ * For now, we will keep the following declarations.
+ */
+template<typename MetricType>
+using FloatHRectBound = HRectBound<MetricType, float>;
+
+template<typename MetricType, typename StatisticType, typename MatType>
+using FloatKDTree = BinarySpaceTree<MetricType, StatisticType, MatType,
+                                    FloatHRectBound, MidpointSplit>;
```

Comment:
I am trusting that you will come back later and refactor the bounds :smile:

Reply:
I have to do so, but I will need some time :+1: